controllers/observabilityendpoint/metrics_collector.go

```diff
@@ -188,6 +188,7 @@ func createDeployment(clusterID string, obsAddonSpec oashared.ObservabilityAddon
 							},
 							VolumeMounts:    mounts,
 							ImagePullPolicy: corev1.PullAlways,
+							Resources:       obsAddonSpec.Resources,
```

Comment:
do we need to check whether obsAddonSpec.Resources is nil?

Reply:
I think we do not need to check because we set a default value for obsAddonSpec. @clyang82 is that right?